src/Microsoft.TestPlatform.Common/Interfaces/Engine/TesthostProtocol/IExecutionManager.cs

```diff
@@ -24,21 +24,23 @@ public interface IExecutionManager
         /// Starts the test run with sources.
         /// </summary>
         /// <param name="adapterSourceMap"> The adapter Source Map.  </param>
+        /// <param name="packages"> The packages which actually contain sources(UWP).</param>
         /// <param name="runSettings"> The run Settings.  </param>
         /// <param name="testExecutionContext"> The test Execution Context. </param>
         /// <param name="testCaseEvents"> EventHandler for handling test cases level events from Engine. </param>
         /// <param name="eventHandler"> EventHandler for handling execution events from Engine. </param>
-        void StartTestRun(Dictionary<string, IEnumerable<string>> adapterSourceMap, string runSettings, TestExecutionContext testExecutionContext, ITestCaseEventsHandler testCaseEvents, ITestRunEventsHandler eventHandler);
+        void StartTestRun(Dictionary<string, IEnumerable<string>> adapterSourceMap, IEnumerable<string> packages, string runSettings, TestExecutionContext testExecutionContext, ITestCaseEventsHandler testCaseEvents, ITestRunEventsHandler eventHandler);
```

Comment:
In case of multiple package, how we maintain pkg to source map or should we only pass one pkg?

Reply:
A testhost will only run for single package, All sources should belong to same package.
In case package is null, that means we don't need to use this info.